lib/facter/resolvers/pciconf.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    class Pciconf < BaseResolver
+      init_resolver
+
+      REGEX_VALUES = { 'VirtualBox' => 'virtualbox', 'XenSource' => 'xenhvm',
+                       'Microsoft Corporation Hyper-V' => 'hyperv', 'Class 8007: Google, Inc' => 'gce',
+                       'VM[wW]are' => 'vmware', '1ab8:' => 'parallels', '[Pp]arallels' => 'parallels',
+                       '(?i)(virtio)' => 'kvm' }.freeze
```

Comment:
This list is a duplicate of the one from the `Lspci` class (`pciconf` is basically the `lspci` of FreeBSD).  Maybe we can avoid duplication but I do not really know the best way to achieve this.

Reply:
over in cloud-init, we use `sysctl -qn kern.vm_guest` and then have this translation table https://github.com/canonical/cloud-init/blob/master/tools/ds-identify#L408

Reply:
also, using sysctl FFI is probably faster than shelling out

Reply:
w00t!  That look like such a better solution than pciconf!

Reply:
I could take some time to work on this PR.  I implemented what you suggested: thanks for pointing-out this sysctl :wink: